### PR TITLE
Add `snakemake --lint`/`snakefmt` support to super-linter

### PR DESCRIPTION
**Pull request by kpj**
This PR implements linting/formatting support for `Snakefiles`.

[Snakemake](https://github.com/snakemake/snakemake/) is a Python-based workflow management system. It is organized in `Snakefiles` which can be linted using [snakemake --lint](https://snakemake.readthedocs.io/en/stable/snakefiles/writing_snakefiles.html#best-practices) and formatted using [snakefmt](https://github.com/snakemake/snakefmt).

### CODE REVIEW

File: lib/linter.sh [hunk 1]
```diff
@@ -1846,6 +1851,28 @@ if [ "${VALIDATE_SHELL_SHFMT}" == "true" ]; then
...
 
+##################
+# SNAKEMAKE LINT #
+##################
+if [ "${VALIDATE_SNAKEMAKE_LINT}" == "true" ]; then
+  ################################
+  # Lint the files with snakefmt #
+  ################################
+  # LintCodebase "FILE_TYPE" "LINTER_NAME" "LINTER_CMD" "FILE_TYPES_REGEX" "FILE_ARRAY"
+  LintCodebase "SNAKEMAKE_LINT" "snakemake --lint" "snakemake --lint -s" ".*\.\(smk\)\$" "\(Snakefile|.*\.smk\)\$" "${FILE_ARRAY_SNAKEMAKE[@]}"
```
Comment (kpj) [inline lib/linter.sh:1862]: Is it safe to assume that `LINTER_COMMAND` will always be followed by the source file?
The `-s` parameter requires this.

File: lib/linter.sh [hunk 2]
```diff
@@ -1859,7 +1859,7 @@ if [ "${VALIDATE_SNAKEMAKE_LINT}" == "true" ]; then
   # Lint the files with snakefmt #
   ################################
   # LintCodebase "FILE_TYPE" "LINTER_NAME" "LINTER_CMD" "FILE_TYPES_REGEX" "FILE_ARRAY"
-  LintCodebase "SNAKEMAKE_LINT" "snakemake --lint" "snakemake --lint -s" ".*\.\(smk\)\$" "\(Snakefile|.*\.smk\)\$" "${FILE_ARRAY_SNAKEMAKE[@]}"
+  LintCodebase "SNAKEMAKE_LINT" "snakemake" "snakemake --lint -s" ".*\.\(smk\)\$" "\(Snakefile|.*\.smk\)\$" "${FILE_ARRAY_SNAKEMAKE[@]}"
```
Comment (lydz1209) [inline lib/linter.sh:1862]: fi

**Review by lydz1209 — CHANGES_REQUESTED** [review]

File: lib/linter.sh (hunk 2 above)
Comment (kpj) [reply lib/linter.sh]: Thanks! Fixed it in the next commit.

File: lib/linter.sh (hunk 1 above)
Comment (admiralAwkbar) [reply lib/linter.sh]: @kpj YOu are correct to assume that command will be followed by the file being evaluated

File: lib/linter.sh [hunk 3]
```diff
@@ -118,6 +118,9 @@ R_LINTER_RULES="${DEFAULT_RULES_LOCATION}/${R_FILE_NAME}"
 # Ruby Vars
 RUBY_FILE_NAME="${RUBY_CONFIG_FILE:-.ruby-lint.yml}"            # Name of the file
 RUBY_LINTER_RULES="${DEFAULT_RULES_LOCATION}/${RUBY_FILE_NAME}" # Path to the ruby lint rules
+# Snakemake Vars
+SNAKEMAKE_FILE_NAME="${SNAKEMAKE_CONFIG_FILE:-.snakefmt.toml}"
```
Comment (admiralAwkbar) [inline lib/linter.sh:122]: Can you add the variable to the dockerfile, and to the readme for documentation?

**Comment by admiralAwkbar** [comment]
@kpj This is great work!
just a few things and I think were good to get this merged up

**Comment by admiralAwkbar** [comment]
@kpj I took the liberty to get it done :)

**Review by IAmHughes — APPROVED** [review]

File: lib/linter.sh (hunk 1 above)
Comment (kpj) [reply lib/linter.sh]: Great, thanks!

**Comment by kpj** [comment]
Awesome, thank you very much!

I have a question regarding testing this PR: In the workflow's `linter.yml` I tried both `uses: github/super-linter@master` and `uses: github/super-linter@133eb655ffcd8153daa551e87040883bcbb3e868` (instead of `uses: github/super-linter@v3`) in order to include Snakemake linting.
However, `github/super-linter:v3` keeps being used:
image
Am I doing something obvious wrong?

(Sorry for polluting this PR, if this is not specific to this project I'll go and debug elsewhere 😃 )

**Comment by lydz1209** [comment]
Le mar. 8 sept. 2020, à 05 h 03, lydz1209 <notifications@github.com> a
écrit :

**Pull request merged**